### PR TITLE
add Kubernetes events preset

### PR DESCRIPTION
**Pull request by povilasv**
First pass at implementing the preset, let me know what you think :)

Ref https://github.com/open-telemetry/opentelemetry-helm-charts/issues/381

Tested with:
```
 helm template --debug --set mode=deployment --set "presets.eventsCollection.enabled=true" .
 ```
 and actually installing the chart:
 
 ```
 helm install --set "presets.eventsCollection.enabled=true" --set "mode=deployment" --generate-name .
 ```

### CODE REVIEW

File: charts/opentelemetry-collector/README.md [hunk 1]
```diff
@@ -138,6 +138,25 @@ config:
           - otlphttp
 ```
 
+### Configuration for Kubernetes events
+
+The collector can be used to collect Kubernetes Events as logs.
+This feature is disabled by default. It has the following requirements:
+
+- It requires the [contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib) version
```
Comment (TylerHelmuth) [inline charts/opentelemetry-collector/README.md:146]: Technically any distribution of the collector with the k8s_events receiver should work.

File: charts/opentelemetry-collector/README.md [hunk 2]
```diff
@@ -138,6 +138,25 @@ config:
...
+of the collector image.
+
+To enable this feature, set the  `presets.eventsCollection.enabled` property to `true`.
+Here is an example `values.yaml`:
+
+```yaml
+mode: daemonset
```
Comment (TylerHelmuth) [inline charts/opentelemetry-collector/README.md:153]:
```suggestion
mode: deployment
```

suggesting deployment based on the comments in values.yaml

File: charts/opentelemetry-collector/templates/clusterrole.yaml [hunk 3]
```diff
@@ -9,5 +9,66 @@ metadata:
   annotations:
     {{- toYaml . | nindent 4 }}
   {{- end }}
-rules: {{ toYaml .Values.clusterRole.rules | nindent 2 -}}
+rules:
+  {{- if .Values.clusterRole.create -}}
+  {{ toYaml .Values.clusterRole.rules | nindent 2 -}}
+  {{- end }}
+  {{- if .Values.presets.eventsCollection.enabled}}
```
Comment (TylerHelmuth) [inline charts/opentelemetry-collector/templates/clusterrole.yaml:16]: It is possible that both `.Values.clusterRole.create` and `.Values.presets.eventsCollection.enabled` are `true`, which will result in both `.Values.clusterRole.rules` and these default k8s events rules being used.  that could result in some duplication.

I think `.Values.clusterRole.rules` should take precedence so I think we can use and `else if` here.

**Review by TylerHelmuth — COMMENTED** [review]
Thanks for adding this preset

**Review by TylerHelmuth — DISMISSED** [review]
@povilasv thanks for adding this preset.  @dmitryax @Allex1 please take a look.

**Comment by dmitryax** [comment]
Given that k8s events receiver is in alpha and potentially can be replaced (see https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/14474#issuecomment-1268819633) I'd recommend holding off introducing this preset.

Enabling k8s attributes processor on tracing seems much more important to me. WDYT?

**Comment by TylerHelmuth** [comment]
> Given that k8s events receiver is in alpha and potentially can be replaced (see [open-telemetry/opentelemetry-collector-contrib#14474 (comment)](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/14474#issuecomment-1268819633)) I'd recommend holding off introducing this preset.

Thanks for bringing this to my attention.  Based on the conversation I'd agree with holding off.


> Enabling k8s attributes processor on tracing seems much more important to me.

I too would love to see a preset for this.  Issue is being tracked via #389.

**Pull request closed without merging**